mypy/stubgenc.py

```diff
@@ -673,3 +749,17 @@ def infer_method_sig(name: str, self_var: str | None = None) -> list[ArgSig]:
     if args is None:
         args = [ArgSig(name="*args"), ArgSig(name="**kwargs")]
     return [ArgSig(name=self_var or "self")] + args
+
+
+def infer_method_ret_type(name: str) -> str:
+    if name.startswith("__") and name.endswith("__"):
+        name = name[2:-2]
+        if name in ("float", "bool", "bytes", "int"):
+            return name
+        elif name in ("eq", "ne", "lt", "le", "gt", "ge", "contains"):
```

Comment:
This isn't entirely safe as `__eq__` and co may return arbitrary types, but it's probably good enough for stubgen.

Reply:
Good point.  I added a comment here to this effect.